src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+import {
+    CloudWatchLogsData,
+    CloudWatchLogsGroupInfo,
+    CloudWatchLogsParameters,
+    LogStreamRegistry,
+    filterLogEventsFromUriComponents,
+} from '../registry/logStreamRegistry'
+import { CloudWatchLogsNode } from '../../../src/cloudWatchLogs/explorer/cloudWatchLogsNode'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+
+import { Wizard } from '../../shared/wizards/wizard'
+import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createURIFromArgs } from '../cloudWatchLogsUtils'
+
+export async function searchLogGroup(registry: LogStreamRegistry): Promise<void> {
+    let result: telemetry.Result = 'Succeeded'
+    const regionCode = 'us-west-2'
+    const logGroupNodes = await getLogGroupNodes(regionCode)
+    const response = await new SearchLogGroupWizard(logGroupNodes).run()
+    if (response) {
+        const logGroupInfo: CloudWatchLogsGroupInfo = {
+            groupName: response.logGroup,
+            regionName: regionCode,
+        }
+
+        const parameters: CloudWatchLogsParameters = {
+            limit: registry.configuration.get('limit', 10000),
+            filterPattern: response.filterPattern,
+        }
+
+        const uri = createURIFromArgs(logGroupInfo, parameters)
+        const initialStreamData: CloudWatchLogsData = {
+            data: [],
+            parameters: parameters,
+            busy: false,
+            logGroupInfo: logGroupInfo,
+            retrieveLogsFunction: filterLogEventsFromUriComponents,
+        }
+
+        await registry.registerLog(uri, initialStreamData)
+        const doc = await vscode.workspace.openTextDocument(uri) // calls back into the provider
+        vscode.languages.setTextDocumentLanguage(doc, 'log')
+        await vscode.window.showTextDocument(doc, { preview: false })
+    } else {
+        result = 'Cancelled'
+    }
+    telemetry.recordCloudwatchlogsOpenStream({ result })
+}
+
+function loadLogGroups(logGroups: AWSTreeNodeBase[]): DataQuickPickItem<string>[] {
+    return logGroups.map<DataQuickPickItem<string>>(groupNode => ({
+        label: groupNode.label!,
+        data: groupNode.label!,
+    }))
+}
+
+async function getLogGroupNodes(regionCode: string) {
+    const artificialNode = new CloudWatchLogsNode(regionCode)
+    const logGroupNodes: AWSTreeNodeBase[] = await artificialNode.getChildren()
+    return logGroupNodes
+}
```

Comment:
It seemed like the easiest way to do it to me, but I can definitely see how it isn't the best approach. Its definitely a little "hacky". I just changed this part to use a `CloudWatchLogs` Client and the `AsyncIterableIterator` stuff rather than a fake node.